docs/4.x/archiving-behavior-specification.md

```diff
@@ -0,0 +1,385 @@
+  set to the name of the report. This will be set as the requested report in the ArchiveProcessor\Params instance that is passed
+  to a plugin's Archiver instance. If the archiver supports partial archiving, it will handle it and a partial archive will be
+  created, with the report archive in it only. This must be set in combination with `plugin` and `pluginOnly`.
+
+## core:archive command
+
+_Brief Description of system:_ The `core:archive` command is meant to be run as a cron job periodically archiving data
+based on a queue. The queue that controls what archives get launched is the `archive_invalidations` table. `core:archive`
+will both: insert into this table before processing each site, and then process the entries, launching the core archiving process
+for each valid entry.
+
+### Range Archiving vs All Others
+
+By default when using core:archive, day, week, month & year periods are not processed in the browser. Range periods
+on the other hand are still processed from web requests. This is because we cannot pre-archive every possible permutation
+of dates that users can specify.
+
+Instead when a range period is no longer considered up-to-date (it's TTL expires), we launch archiving for the range.
+Since the range just aggregates day periods, this is still performant, as long as archiving for the days is not launched.
+
+### Launching the core archiving process
+
+The core:archive command launches the core archiving process through new processes, which allows it to achieve concurrency.
+If it's supported by the OS configuration and PHP runtime configuration, these processes are CLI processes spawned through
+`shell_exec()` running the `climulti:request` command. The `climulti:request` command takes a query string and processes it
+like it is an HTTP request.
+
+If it's not supported, then we use CURL multi requests. CURL is not as fast as launching cli processes, so we prefer to use
+`climulti:request`.
+
+In each case, we call the `CoreAdminHome.archiveReports` API method which launches core archiving.
+Note: previously the core:archive command would query `API.get` which would implicitly trigger the archiving process.
+
+The request URI will always have `trigger=archivephp` which, when superuser access is granted as is the case during archiving,
+will force the archiving process to initiate.
+
+**Batching Archive Jobs**
+
+In the core:archive command, we try and process multiple archives at the same time to make the process more performant. The
+number of concurrent processes to run at a time is specified by the `--concurrent-requests-per-website` CLI option.
+
+Archive jobs for different sites are never run concurrently from within the same `core:archive` process. Sites can be
+archived concurrently but this is only possible by running multiple `core:archive` commands.
+
+**Order of Execution**
+
+Some archive invalidations have to be handled before others. Specifically:
```

Comment:
could mention here the "why". Eg all visits might be executed first because it is viewed by default and looked at most of the time so this data is visible in the UI sooner. Also more recent dates are viewed more often compared to older dates which is why we archive these first

Reply:
Was it also to avoid some unpredictability in archiving?